internal/runner.js

```diff
@@ -123,17 +124,33 @@ function compile(rawArgs) {
             return true;
           },
           e => {
-            console.warn('Error in processing. Args:', args);
-            console.warn('Error:', e);
+            worker.log('Error in processing. Args:', args);
+            worker.log('Error:', e);
             return false;
           });
 }
 
-const args = process.argv.slice(2);
-compile(args).then(
-    success => {
-      // Arbitrary exit code for failure, otherwise we exit succesfully.
-      if (!success) {
-        process.exit(2);
-      }
-    });
+if (worker.runAsWorker(process.argv)) {
+  worker.log('PostCSS runner starting as worker...');
+  worker.runWorkerLoop(compile);
```

Comment:
The `compile()` function includes loading the plugins, which can be somewhat expensive. Could you load them outside the worker loop instead?

Reply:
Thanks for catching this - done.

Reply:
I was thinking that you could cache the entire plugin, not just the module—I believe modules are already cached by Node's `require` infrastructure. That said, thinking more about it I suspect caching instantiated modules is more complex than is worth implementing in this PR.

Reply:
You're right, undone the change. `runfiles.resolve` is technically something that wouldn't get cached but I suspect it's too small of a micro-optimisation (that potentially adds overhead).

I do think it's worth looking into caching instantiated modules but I agree it would get complex. Particularly because they would need to be keyed by the plugin options somehow.

Reply:
Or some design where `postcss_stack` would be setting up the plugin instances once, and the worker would just take many `postcss_binary` calls.

I've renamed the PR to clarify it's only rudimentary worker support that's added, and actual gains from moving to workers would need to be realised in followup PRs.